### ISSUE TITLE
Fix nested nth root

### ISSUE THREAD

**Issue by petrovich2k** [mathjax/MathJax#2870]
### Issue Summary

When I convert a nth root which index is also nthroot, I get a corrupted SVG. See below what I mean.
![nth-root-summary](https://user-images.githubusercontent.com/72270591/168701826-65d77fe0-8058-4588-9078-25913b6f9aa6.png)

Sample TeX: `\\sqrt[\\sqrt[]{}]{}`, `\\sqrt[\\sqrt[3]{8}]{16}`

Actually, I use [tex-to-svg](https://github.com/CreatifCreateur/TeX-to-SVG) to convert TeX fo SVG. It references to MathJax and executes the following code.
```
    const tex = new TeX({ packages });
    const svg = new SVG({ fontCache: (FONT_CACHE ? 'local' : 'none') });
    const html = mathjax.document('', { InputJax: tex, OutputJax: svg });
    const node = html.convert(str, {
        display: !INLINE,
        em: options.em,
        ex: options.ex,
        containerWidth: options.width
    });
```

The error occurs in TexParser. It says "Could not find closing ']'". 

Call stack
```
./node_modules/mathjax-full/js/input/tex/TexParser.js.TexParser.GetBrackets (TexParser.js:259)
./node_modules/mathjax-full/js/input/tex/base/BaseMethods.js.BaseMethods.Sqrt (BaseMethods.js:273)
./node_modules/mathjax-full/js/input/tex/SymbolMap.js.CommandMap.parse (SymbolMap.js:221)
./node_modules/mathjax-full/js/input/tex/MapHandler.js.SubHandler.parse (MapHandler.js:78)
./node_modules/mathjax-full/js/input/tex/TexParser.js.TexParser.parse (TexParser.js:104)
controlSequence (ParseMethods.js:59)
./node_modules/mathjax-full/js/input/tex/SymbolMap.js.AbstractSymbolMap.parse (SymbolMap.js:74)
./node_modules/mathjax-full/js/input/tex/MapHandler.js.SubHandler.parse (MapHandler.js:78)
./node_modules/mathjax-full/js/input/tex/TexParser.js.TexParser.parse (TexParser.js:104)
./node_modules/mathjax-full/js/input/tex/TexParser.js.TexParser.Parse (TexParser.js:136)
TexParser (TexParser.js:69)
./node_modules/mathjax-full/js/input/tex.js.TeX.compile (tex.js:115)
./node_modules/mathjax-full/js/core/MathItem.js.AbstractMathItem.compile (MathItem.js:56)
(anonymous) (MathDocument.js:115)
./node_modules/mathjax-full/js/core/MathDocument.js.RenderList.renderConvert (MathDocument.js:167)
./node_modules/mathjax-full/js/core/MathItem.js.AbstractMathItem.convert (MathItem.js:52)
./node_modules/mathjax-full/js/core/MathDocument.js.AbstractMathDocument.convert (MathDocument.js:325)
TeXToSVG (TeXToSVG.js:30)
```

### Steps to Reproduce:

1. download [the repository](https://github.com/petrovich2k/nested-nth-root-demo)
2. `npm install`
3. `npm run build`
4. serve the dist folder (`npm run start` or use live server)

### Technical details:

* MathJax Version: 3.2

### Supporting information:

I have created two repositories. [The first one](https://github.com/petrovich2k/nested-nth-root-demo) reproduces the issue. You will see sth like that.
![nth-root-issue](https://user-images.githubusercontent.com/72270591/168702204-2d72207f-8500-47a9-8e01-edf21025f02b.png)

[The second one](https://github.com/petrovich2k/nested-nth-root-demo-fixed) shows that my fix works well.
![nth-root-fix](https://user-images.githubusercontent.com/72270591/168702268-734a0d13-9153-4624-9a74-24c34647067d.png)

Please download them and execute the following commands
1. `npm install`
2. `npm run build`
3. serve the dist folder (`npm run start` or use live server)

I have also opened [a pull-request for my solution](https://github.com/mathjax/MathJax-src/pull/812).

**Comment by dpvc** [comment]
Neither `\sqrt[\sqrt[]{}]{}`, `\sqrt[\sqrt[3]{8}]{16}` is valid LaTeX, and both will report an error in actual LaTeX.  The error message in LaTeX is `Argument of \@sqrt has an extra }`, but that is due to the same issue:  LaTeX doesn't process matched brackets, so in `\sqrt[\sqrt[3]{8}]{16}`, the contents of the backet is `\sqrt[3` as the root and the radicand is `8` (followed by the characters `]{16}` which are not part of any root).  When `\sqrt[3` is processed, the `[` is unclosed, and (due to the macro definition in LaTeX, where it is processed as `{\sqrt[3}` somewhere), TeX reports the `}` as an extra brace rather than the missing bracket that MathJax reports.  But those are effectively the same issue.

So MathJax's processing is actually correct, here.  The proper LaTeX for what you are trying to do is `\sqrt[{\sqrt[3]{8}}]{16}`, with extra braces inside the brackets for the first `\sqrt`; that is the standard LaTeX approach to hiding the nested brackets.

**Comment by dpvc** [comment]
PS, there *is* a display issue in the SVG output for `\sqrt[{\sqrt[3]{8}}]{16}`, but that was reported in #2764 and resolved in PR mathjax/MathJax-src#743.

**Comment by petrovich2k** [comment]
Thanks, @dpvc. I didn't know that the nested nth root can be wrapped by curly brackets. I've found another solution based on regular expressions without amending the MathJax repository.
